### PR TITLE
Add a custom random number generator utility

### PR DESCRIPTION
**Pull request by andrewsbrown**
See http://redmine.named-data.net/issues/3500

### CODE REVIEW

File: src/net/named_data/jndn/util/Common.java [hunk 1]
```diff
@@ -36,6 +38,36 @@
...
+   */
+  public static synchronized void
+  setRandom(Random randomNumberGenerator) {
+    if(randomNumberGenerator_ == null) {
+      randomNumberGenerator_ = randomNumberGenerator;
+    }
+    else{
+      throw new UnsupportedOperationException("The random number generator may only be set once");
+    }
+  }
```
Comment (andrewsbrown) [inline src/net/named_data/jndn/util/Common.java]: @cawka, take a look at this: in this scenario the user programmatically sets this and can only do so once

File: src/net/named_data/jndn/encoding/Tlv0_1_1WireFormat.java [hunk 2]
```diff
@@ -117,7 +117,7 @@
     {
       // This is the most common case. Generate a nonce.
       ByteBuffer nonce = ByteBuffer.allocate(4);
-      random_.nextBytes(nonce.array());
+      Common.getRandom().nextBytes(nonce.array());
```
Comment (cawka) [inline src/net/named_data/jndn/encoding/Tlv0_1_1WireFormat.java]: nonce fields does not warrant "Secure" random

File: src/net/named_data/jndn/Interest.java [hunk 3]
```diff
@@ -619,7 +619,7 @@ else if (!linkWireEncoding_.isNull()) {
 
     ByteBuffer newNonce = ByteBuffer.allocate(currentNonce.size());
     while (true) {
-      random_.nextBytes(newNonce.array());
+      Common.getRandom().nextBytes(newNonce.array());
```
Comment (cawka) [inline src/net/named_data/jndn/Interest.java]: shouldn't be "secure" random

Comment (andrewsbrown) [inline src/net/named_data/jndn/Interest.java]: @jefft0 do you want to keep a plain `Random` reference in here?

File: src/net/named_data/jndn/encoding/Tlv0_1_1WireFormat.java (hunk 2 above)
Comment (andrewsbrown) [inline src/net/named_data/jndn/encoding/Tlv0_1_1WireFormat.java]: @jefft0 do you want to keep a plain `Random` reference in here?

File: src/net/named_data/jndn/Interest.java (hunk 3 above)
Comment (jefft0) [inline src/net/named_data/jndn/Interest.java]: Sure. In Java, the default Random() constructor claims that it seeds to something unique (unlike other platforms).

File: src/net/named_data/jndn/encoding/Tlv0_1_1WireFormat.java (hunk 2 above)
Comment (jefft0) [inline src/net/named_data/jndn/encoding/Tlv0_1_1WireFormat.java]: Sure. In Java, the default Random() constructor claims that it seeds to something unique (unlike other platforms).

**Pull request merged**